www/js/con-student.js

```diff
@@ -0,0 +1,352 @@
+		 These don't need to be recorded in the model, since they
+		 are applied each time the node editor is opened.
+		 */
+		initialViewSettings: function(type){
+			//make display none for all fields initially
+			var qtyElements = ["descriptionInputboxContainerStudent","variableTypeContainer",
+			"variableInputboxContainerStudent","valueUnitsContainer"];
+			var eqElements = ["descriptionInputboxContainerStudent","expressionDiv"];
+		
+			if(type == "quantity"){
+				eqElements.forEach(function(elem){
+					console.log("element",elem);
+					style.set(elem,"display","none");
+				});
+				qtyElements.forEach(function(elem){
+					console.log("element",elem);
+					style.set(elem,"display","block");
+				});
+			}else if(type == "equation"){
+				qtyElements.forEach(function(elem){	
+					console.log("element",elem);
+					style.set(elem,"display","none");
+				});
+				eqElements.forEach(function(elem){
+					console.log("element",elem);
+					style.set(elem,"display","block");
+				});
+
+			}
+			//emptying message box when a node is open
+			console.log("emptying message");
+			dojo.empty("messageOutputbox");
+		},
+		initialControlSettings: function (nodeid) {
+			// Apply settings from PM
+			console.log("Initial Control Settings for Student");
+			
+			// Set the selected value in the description.
+			var desc = this._model.student.getAuthoredID(nodeid);
+
+			console.log('description is', desc || "not set");
+			
+			registry.byId(this.controlMap.description).set('value', desc || 'defaultSelect', false);
+			
+			registry.byId(this.controlMap.variable).set('value', desc || 'defaultSelect', false);
+
+			var _variableTypes = ["unknown","parameter","dynamic"];
+			array.forEach(_variableTypes, function(_type){
+				registry.byId(_type+"Type").set('checked',false);
+				registry.byId(_type+"Type").set('disabled',false);
+			})
+			
+			style.set('valueInputboxContainer','display','none');
+			style.set('valueInputboxContainer','disabled',false);
+			
+			var _variableType = this._model.student.getVariableType(nodeid);
+			if(_variableType){
+				var _returnObj = this._PM.getActionForType(nodeid, _variableType);
+				
+				array.forEach(_returnObj, function(directive) {
+					if(directive.attribute == "status" && directive.value == "correct"){
+						registry.byId(_variableType+"Type").set('checked','checked');
+						array.forEach(_variableTypes, function(_type){
+							if(_type !== _variableType){
+								registry.byId(_type+"Type").set('disabled',true);
+							}
+						})
+						if(_variableType =="parameter" || _variableType == "dynamic"){
+							style.set('valueInputboxContainer','display','block');
+						}else{
+							style.set('valueInputboxContainer','display','none');
+							style.set('valueInputboxContainer','disabled',false);
+						}
+					}
+				});
+			}			
+			/*
+			 Set color and enable/disable
+			 */
+			array.forEach(this._model.student.getStatusDirectives(nodeid), function (directive) {
```

Comment:
is there a reason for this to be here? As far as I remember this was done in apply status directive function.